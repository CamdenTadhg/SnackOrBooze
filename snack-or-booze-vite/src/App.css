#root {
  margin: 0 auto;
  text-align: center;
}

.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
  transition: filter 300ms;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
}

main{
  width: 100vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 5vh;
  height: calc(100vh - 56px);
  overflow: scroll;
  color: white;
}

main a{
  color: white;
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}

p {
  color: white;
}
