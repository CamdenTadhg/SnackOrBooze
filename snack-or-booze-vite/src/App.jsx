import React, { useState, useEffect } from "react";
import { BrowserRouter } from "react-router-dom";
import "./App.css";
import Home from "./Home";
import SnackOrBoozeApi from "./Api";
import NavBar from "./NavBar";
import { Route, Routes } from "react-router-dom";
import Menu from "./Menu";
import MenuItem from "./MenuItem";
import AddForm from './AddForm';
import NotFound from './NotFound';
import 'bootstrap/dist/css/bootstrap.min.css';

//renders loading screen if loading, else renders app. 
function App() {
  const [isLoading, setIsLoading] = useState(true);
  const [snacks, setSnacks] = useState([]);
  const [drinks, setDrinks] = useState([]);

  useEffect(() => {
    //get the list of snacks from the API and set state appropriately
    async function getSnacks() {
      console.log('starting getSnacks');
      setIsLoading(true);
      let loadSnacks = await SnackOrBoozeApi.getGoodies('snacks');
      console.log('getSnacks data returned')
      setSnacks(loadSnacks);
      setIsLoading(false);
    }
    //get the list of drinks from the API and set state appropriately
    async function getDrinks() {
      console.log('starting getDrinks');
      setIsLoading(true);
      let loadDrinks = await SnackOrBoozeApi.getGoodies('drinks');
      console.log('getDRinks data returned');
      setDrinks(loadDrinks);
      setIsLoading(false);
    }
    getSnacks();
    getDrinks();
  }, []);

  if (isLoading) {
    return <p>Loading &hellip;</p>;
  }

  return (
    <div className="App">
      <BrowserRouter>
        <NavBar />
        <main>
          <Routes>
            <Route path="/" element={<Home snacks={snacks} drinks={drinks}/>} />
            <Route path="/snacks" element={<Menu items={snacks} type="snacks" />} />
            <Route path="/snacks/:id" element={<MenuItem items={snacks} cantFind="/snacks" />}/>
            <Route path="/drinks" element={<Menu items={drinks} type="drinks" />} />
            <Route path='/drinks/:id' element={<MenuItem items={drinks} cantFind="/drinks" />}/>
            <Route path='/add' element={<AddForm setSnacks={setSnacks} setDrinks={setDrinks}/>}/>
            <Route path='*' element={<NotFound />}/>
          </Routes>
        </main>
      </BrowserRouter>
    </div>
  );
}

export default App;

//create Snack or Booze website
//get tests running
  //App.test.jsx (1.5 hours)
  //Submit

