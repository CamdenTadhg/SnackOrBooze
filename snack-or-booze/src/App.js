import React, { useState, useEffect } from "react";
import { BrowserRouter } from "react-router-dom";
import "./App.css";
import Home from "./Home";
import SnackOrBoozeApi from "./Api";
import NavBar from "./NavBar";
import { Route, Switch } from "react-router-dom";
import Menu from "./Menu";
import MenuItem from "./MenuItem";
import AddForm from './AddForm';
import NotFound from './NotFound';

function App() {
  const [isLoading, setIsLoading] = useState(true);
  const [snacks, setSnacks] = useState([]);
  const [drinks, setDrinks] = useState([]);

  useEffect(() => {
    async function getSnacks() {
      setIsLoading(true);
      let snacks = await SnackOrBoozeApi.getGoodies('snacks');
      setSnacks(snacks);
      setIsLoading(false);
    }
    async function getDrinks() {
      setIsLoading(true);
      let drinks = await SnackOrBoozeApi.getGoodies('drinks');
      setDrinks(drinks);
      setIsLoading(false);
    }
    getSnacks();
    getDrinks();
  }, []);

  if (isLoading) {
    return <p>Loading &hellip;</p>;
  }

  return (
    <div className="App">
      <BrowserRouter>
        <NavBar />
        <main>
          <Switch>
            <Route exact path="/">
              <Home snacks={snacks} drinks={drinks}/>
            </Route>
            <Route exact path="/snacks">
              <Menu items={snacks} type="snacks" />
            </Route>
            <Route path="/snacks/:id">
              <MenuItem items={snacks} cantFind="/snacks" />
            </Route>
            <Route exact path="/drinks">
              <Menu items={drinks} type="drinks" />
            </Route>
            <Route path='/drinks/:id'>
              <MenuItem items={drinks} cantFind="/drinks" />
            </Route>
            <Route path='/add'>
              <AddForm setSnacks={setSnacks} setDrinks={setDrinks}/>
            </Route>
            <Route>
              <NotFound />
            </Route>
          </Switch>
        </main>
      </BrowserRouter>
    </div>
  );
}

export default App;

//create Snack or Booze website
  //get tests running
//Rewrite tests on Vite version
  //write smoke tests
  //write snapshop tests
  //write api tests
  //write content tests
  //write event tests
//Fix styling on vite version