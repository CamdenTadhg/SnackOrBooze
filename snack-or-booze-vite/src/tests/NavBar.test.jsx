import {render} from '@testing-library/react';
import {expect, test} from 'vitest';
import NavBar from '../NavBar';
import App from '../App';
import {MemoryRouter} from 'react-router-dom';

test('renders the NavBar component', () => {
    render(
        <MemoryRouter>
            <NavBar />
        </MemoryRouter>);
});

test('matches snapshot', () => {
    const navBar = render(
        <MemoryRouter>
            <NavBar />
        </MemoryRouter>);
    expect(navBar).toMatchSnapshot();
});

test('renders the right content', async () => {
    const {getByText} = render(
        <MemoryRouter>
            <NavBar />
        </MemoryRouter>);
    expect(getByText(/Add to Menu/i)).toBeInTheDocument();
})