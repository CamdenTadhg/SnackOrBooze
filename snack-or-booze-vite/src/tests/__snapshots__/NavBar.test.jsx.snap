// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="navbar navbar-expand-md"
        >
          <div
            class="container-fluid"
          >
            <a
              aria-current="page"
              class="navbar-brand active"
              href="/"
            >
              Snack or Booze
            </a>
            <ul
              class="ml-auto navbar-nav"
            >
              <li
                class="nav-item"
              >
                <a
                  class=""
                  href="/snacks"
                >
                  Snacks
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class=""
                  href="/drinks"
                >
                  Drinks
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class=""
                  href="/add"
                >
                  Add to Menu
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="navbar navbar-expand-md"
      >
        <div
          class="container-fluid"
        >
          <a
            aria-current="page"
            class="navbar-brand active"
            href="/"
          >
            Snack or Booze
          </a>
          <ul
            class="ml-auto navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                class=""
                href="/snacks"
              >
                Snacks
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class=""
                href="/drinks"
              >
                Drinks
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class=""
                href="/add"
              >
                Add to Menu
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
